src/HeaderSecurity.php

```diff
@@ -129,6 +129,12 @@ public static function isValid($value)
      */
     public static function assertValid($value)
     {
+        if (! is_string($value)) {
```

Comment:
This is somewhat problematic. Numeric values are allowed in headers — e.g., Content-Length — which means asserting that the value is a string could lead to issues for existing users (including those using the `SapiStreamEmitter`).

That said, we currently enforce this when an array of header values is passed to `withHeader()` (via the private `arrayContainsOnlyStrings()` method).

My feeling is that we should likely cast numeric values to strings when storing headers, vs. throwing them out outright. This would, of course, necessitate further changes from what you were already planning to introduce in this pull request.